src/library_wasmfs_opfs.js

```diff
@@ -14,14 +14,65 @@ mergeInto(LibraryManager.library, {
   $wasmfsOPFSBlobs__deps: ["$HandleAllocator"],
   $wasmfsOPFSBlobs: "new HandleAllocator()",
 
-  _wasmfs_opfs_init_root_directory__deps: ['$wasmfsOPFSDirectoryHandles'],
+#if !USE_PTHREADS
+  $wasmfsFileSystemAsyncAccessHandle: class FileSystemAsyncAccessHandle {
+    // This class implements the same interface as the sync version, but has
+    // async reads and writes. Hopefully this will one day be implemented by the
+    // platform so we can remove it.
+    constructor(handle) {
+      this.handle = handle;
+    }
+    async close() {}
+    async flush() {}
+    async getSize() {
+      let file = await this.handle.getFile();
+      return file.size;
+    }
+    async read(buffer, options = { at: 0 }) {
+      let file = await this.handle.getFile();
+      let slice = await file.slice(options.at);
```

Comment:
Should read and write take size arguments as well? It looks like a tiny read at the beginning of the file will still end up reading the whole file right now.

Reply:
I added the end position of the slice. 

As for read/write taking size, isn't the size of the read just the buffer size? Also, the sync versions of the API don't take size or maybe I'm misunderstanding the question.

Reply:
Oh no, I just forgot that the buffer knows its own size in JS.